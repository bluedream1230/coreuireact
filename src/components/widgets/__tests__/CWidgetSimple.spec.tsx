import * as React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { CWidgetSimple } from '../../../index'

test('loads and displays CWidgetSimple component', async () => {
  const { container } = render(<CWidgetSimple/>)
  expect(container).toMatchSnapshot()
})

test('CWidgetSimple customize', async () => {
    const { container } = render(
        <CWidgetSimple
            className="bazinga"
            title="title"
            value="value"
        >
            Test
        </CWidgetSimple>
    )
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('bazinga')
    expect(container.firstChild).toHaveClass('card')
    if(container.firstChild === null || container.firstChild.firstChild === null){
        expect(true).toBe(false)
    }else{
        expect(container.firstChild.firstChild).toHaveClass('card-body')
        expect(container.firstChild.firstChild.firstChild).toHaveClass('text-medium-emphasis')
        expect(container.firstChild.firstChild.firstChild).toHaveClass('small')
        expect(container.firstChild.firstChild.firstChild).toHaveClass('text-uppercase')
        expect(container.firstChild.firstChild.firstChild).toHaveClass('fw-semibold')
        expect(container.firstChild.firstChild.firstChild).toHaveTextContent('title')
        expect(container.firstChild.firstChild.lastChild).toHaveTextContent('Test')
    }
})