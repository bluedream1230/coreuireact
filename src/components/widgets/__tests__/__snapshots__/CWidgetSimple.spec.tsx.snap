// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CWidgetSimple customize 1`] = `
<div>
  <div
    class="card bazinga"
  >
    <div
      class="card-body text-center"
    >
      <div
        class="text-medium-emphasis small text-uppercase fw-semibold"
      >
        title
      </div>
      <div
        class="fs-6 fw-semibold py-3"
      >
        value
      </div>
      Test
    </div>
  </div>
</div>
`;

exports[`loads and displays CWidgetSimple component 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-body text-center"
    />
  </div>
</div>
`;
